src/main/java/com/fouristhenumber/utilitiesinexcess/common/blocks/BlockDrum.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    @Override
+    public void breakBlock(World world, int x, int y, int z, Block block, int meta) {
+
+        ItemStack drop = new ItemStack(this, 1, meta);
+
+        TileEntity tile = world.getTileEntity(x, y, z);
+        if (tile instanceof TileEntityDrum drum) {
+            if (drum.tank.getFluid() != null) {
+                ItemBlockDrum.setFluid(
+                    drop,
+                    drum.tank.getFluid()
+                        .copy());
+            } else {
+                ItemBlockDrum.clearFluid(drop);
+            }
+        }
+
+        float f = 0.7F;
+        double dx = (double) (world.rand.nextFloat() * f) + (double) (1.0F - f) * 0.5D;
+        double dy = (double) (world.rand.nextFloat() * f) + (double) (1.0F - f) * 0.5D;
+        double dz = (double) (world.rand.nextFloat() * f) + (double) (1.0F - f) * 0.5D;
+
+        EntityItem entityItem = new EntityItem(world, x + dx, y + dy, z + dz, drop);
+        world.spawnEntityInWorld(entityItem);
+        world.removeTileEntity(x, y, z);
+        super.breakBlock(world, x, y, z, block, meta);
+    }
+
+    public static class ItemBlockDrum extends ItemBlock implements IFluidContainerItem {
+
+        public static int capacity = -1;
+
+        public ItemBlockDrum(Block block) {
+            super(block);
+            this.capacity = ((BlockDrum) block).capacity;
+        }
+
+        @Override
+        public FluidStack getFluid(ItemStack stack) {
+            return getFluidFromStack(stack);
+        }
+
+        @Override
+        public int getCapacity(ItemStack stack) {
+            return capacity;
+        }
+
+        @Override
+        public int fill(ItemStack stack, FluidStack resource, boolean doFill) {
+            if (resource == null) {
+                return 0;
+            }
+
+            FluidStack currentFluid = getFluid(stack);
+
+            if (currentFluid == null) {
+                int fillAmount = Math.min(capacity, resource.amount);
+                if (doFill && fillAmount > 0) {
+                    FluidStack newFluid = resource.copy();
+                    newFluid.amount = fillAmount;
+                    setFluid(stack, newFluid);
+                }
+                return fillAmount;
+            } else {
+                if (!currentFluid.isFluidEqual(resource)) {
+                    return 0;
+                }
+
+                int space = capacity - currentFluid.amount;
+                if (space <= 0) {
+                    return 0;
+                }
+
+                int fillAmount = Math.min(space, resource.amount);
+                if (doFill && fillAmount > 0) {
+                    currentFluid.amount += fillAmount;
+                    setFluid(stack, currentFluid);
+                }
+                return fillAmount;
+            }
+        }
+
+        @Override
+        public FluidStack drain(ItemStack stack, int maxDrain, boolean doDrain) {
+            FluidStack currentFluid = getFluid(stack);
```

Comment:
Same here, this could cause issues with stackable fluid containers

Reply:
see other comment